src/zip.ts

```diff
@@ -100,10 +101,14 @@ export async function* loadFiles(files: ForAwaitable<ZipFileDescription & Metada
   yield makeUint8Array(end)
 }
 
-export function fileHeader(file: ZipFileDescription & Metadata) {
+export function fileHeader(file: ZipFileDescription & Metadata, useLanguageEncodingFlag: boolean = true) {
   const header = makeBuffer(fileHeaderLength)
   header.setUint32(0, fileHeaderSignature)
-  header.setUint32(4, 0x2d_00_0800) // ZIP version 4.5 | flags, bit 3 on = size and CRCs will be zero
+  header.setUint16(4, 0x2d_00) // ZIP version 4.5 
+  if (file.isEncodedNameUtf8 && useLanguageEncodingFlag)
+    header.setUint16(6, 0x0808) // flags, bit 3 on = size and CRCs will be zero, bit 11 on = filename is utf-8
+  else
+    header.setUint16(6, 0x0800) // flags, bit 3 on
```

Comment:
I know, I said I don't mind different code styles, but this feels really verbose for something so small. Maybe condense it with a ternary of even a bitwise OR (we are dealing with bit flags after all) ? Like this : 
```suggestion
  header.setUint16(4, 0x2d_00) // ZIP version 4.5
  // flags, bit 3 on = size and CRCs will be zero, bit 11 on if filename is utf-8
  header.setUint16(6, 0x0800 | (file.isEncodedNameUtf8 && useLanguageEncodingFlag && 8))
```